src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -43,19 +45,14 @@ public void Apply(BsonClassMap classMap)
             }
 
             var anyConstructorsWereFound = false;
-            var constructorBindingFlags = BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance;
-            foreach (var ctor in typeInfo.GetConstructors(constructorBindingFlags))
+            foreach (var ctor in typeInfo.GetConstructors(_constructorBindingFlags))
```

Comment:
I find moving the binding flag constants to a non-local variable (field in this case) unhelpful.

It means we can't fully understand what line 48 does without tracking down what the *current* value of the variable is.

I would never write code this way but I am willing to LGTM this way if you insist.

Reply:
I would say it's helpful, it means that all searching for constructors in this file work in the same way regardless of what lines in this file we're looking in

Reply:
if you think that's not informative, we can change the current variable name, but I would say it's not necessary

Reply:
done